MSAL/test/app/mac/MSALAcquireTokenViewController.m

```diff
@@ -0,0 +1,366 @@
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+//------------------------------------------------------------------------------
+
+#import "MSALAcquireTokenViewController.h"
+#import <MSAL/MSAL.h>
+#import "MSALTestAppSettings.h"
+#import "MSALScopesViewController.h"
+#import "MSALInteractiveTokenParameters.h"
+#import "MSALPublicClientApplication+Internal.h"
+#import "MSIDDefaultTokenCacheAccessor.h"
+#import "MSALSilentTokenParameters.h"
+#import "WebKit/WebKit.h"
+
+static NSString * const clientId = @"clientId";
+static NSString * const redirectUri = @"redirectUri";
+static NSString * const defaultScope = @"User.Read";
+
+@interface MSALAcquireTokenViewController ()
+
+@property (weak) IBOutlet NSPopUpButton *profiles;
+@property (weak) IBOutlet NSTextField *clientId;
+@property (weak) IBOutlet NSTextField *redirectUri;
+@property (weak) IBOutlet NSTextField *scopesLabel;
+@property (weak) IBOutlet NSSegmentedControl *promptBehavior;
+@property (weak) IBOutlet NSTextField *loginHintField;
+@property (weak) IBOutlet NSTextView *resultView;
+@property (weak) IBOutlet NSTextField *extraQueryParamsField;
+@property (weak) IBOutlet NSSegmentedControl *webViewType;
+@property (weak) IBOutlet NSSegmentedControl *validateAuthority;
+@property (weak) IBOutlet NSStackView *acquireTokenView;
+@property (weak) IBOutlet WKWebView *webView;
+
+
+@property MSALTestAppSettings *settings;
+@property NSArray *selectedScopes;
+
+@end
+
+@implementation MSALAcquireTokenViewController
+
+- (void)viewDidLoad {
+    [super viewDidLoad];
+    self.settings = [MSALTestAppSettings settings];
+    [self populateProfiles];
+    self.selectedScopes = @[defaultScope];
+}
+
+- (void)populateProfiles
+{
+    [self.profiles removeAllItems];
+    [self.profiles setTarget:self];
+    [self.profiles setAction:@selector(selectedProfileChanged:)];
+    [[MSALTestAppSettings profiles] enumerateKeysAndObjectsUsingBlock: ^(id key, id obj, BOOL *stop) {
+        [self.profiles addItemWithTitle:key];
+    }];
+    
+    [self.profiles selectItemWithTitle:[MSALTestAppSettings currentProfileName]];
+    self.clientId.stringValue = [[MSALTestAppSettings currentProfile] objectForKey:clientId];
+    self.redirectUri.stringValue = [[MSALTestAppSettings currentProfile] objectForKey:redirectUri];
+}
+
+- (IBAction)selectedProfileChanged:(id)sender
+{
+    [self.settings setCurrentProfile:[self.profiles indexOfSelectedItem]];
+    self.clientId.stringValue = [[MSALTestAppSettings currentProfile] objectForKey:clientId];
+    self.redirectUri.stringValue = [[MSALTestAppSettings currentProfile] objectForKey:redirectUri];
+}
+
+- (void)prepareForSegue:(NSStoryboardSegue *)segue sender:(id)sender
+{
+    if ([segue.identifier isEqualToString:@"addScopesSegue"])
+    {
+        MSALScopesViewController *scopesController = (MSALScopesViewController *)segue.destinationController;
+        scopesController.delegate = self;
```

Comment:
nit: should scopes be updated through the profile instead?

Reply:
The profiles from the config file does not have scopes as part of it yet. We can add that in the future.

Reply:
Yes, but we could have settings object hold the currently stored scopes, right?

Reply:
Sure we can have that but we can do that in a later PR by adding the scopes to NSUserDefaults. This is just a first revision.